### ISSUE TITLE
[Android][EMULATOR] KeyboardAccessoryView is detached from keyboard when input is focused

### ISSUE THREAD

**Issue by theohdv** [ardaogulcan/react-native-keyboard-accessory#17]
It works fine on iOS.

**Structure of my screen:**
```
<View style={{ flex: 1 }}>
  <View>
    ...
  </View>
  <KeyboardAccessoryView>
    <View><Text> ... </Text></View>
  </KeyboardAccessoryView>
</View>
```

**The result:**
<img width="464" alt="screen shot 2018-06-19 at 16 09 28" src="https://user-images.githubusercontent.com/26060802/41602772-d89d23b4-73db-11e8-821f-d63cbd8d6a43.png">
<img width="462" alt="screen shot 2018-06-19 at 16 09 38" src="https://user-images.githubusercontent.com/26060802/41602783-dac6eda0-73db-11e8-92ea-26ec4eb6f78d.png">

**Comment by ardaogulcan** [comment]
Hi,

Is your app is Expo ? Otherwise, please check your `android:windowSoftInputMode` settings on `android/app/src/main/AndroidManifest.xml`. It should be "adjustResize" or "adjustPan".

**Comment by theohdv** [comment]
No I don't use expo. I was using `adjustPan` and the `TextInput` was closer but still detached. And when I remove completely `android:windowSoftInputMode` property I have the screenshot I attached to this issue.

**Comment by kimho** [comment]
I solved it like below.
But it's temporary and not graceful.

```
_debugKeyboardInput=()=>{
    let tmpComment = this.state.comment;
    this.setState({ comment: tmpComment + "." });
    setTimeout(() => {
      this.setState({ comment: tmpComment });
    }, 100);
}
...
<Input
  onFocus={()=>this._debugKeyboardInput()}
...
/>
```

**Comment by ardaogulcan** [comment]
Hi @theohdv,

Now I can reproduce your case, thing u need is a wrapper `View` with `flex: 1` if you are not using ScrollView before `KeyboardAccessoryView. Check the below code.

```js
import React, { Component } from 'react';

import {
  StyleSheet,
  View,
  TextInput,
  Button,
  Text,
} from 'react-native';

import { KeyboardAccessoryView } from '../react-native-keyboard-accessory';

class ViewExample extends Component {
  render() {
    return (
      <View style={styles.container}>
        <View style={{ flex: 1 }}>
          <View style={{ height: 250, backgroundColor: 'white' }}>
            <Text>Test View</Text>
          </View>
        </View>
        <KeyboardAccessoryView alwaysVisible={true}>
          <View style={styles.textInputView}>
            <TextInput
              underlineColorAndroid="transparent"
              style={styles.textInput}
              multiline={true} />
            <Button
              style={styles.textInputButton}
              title="Send"
              onPress={() => {}}/>
          </View>
        </KeyboardAccessoryView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  textInputView: {
    padding: 8,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  textInput: {
    flexGrow: 1,
    borderWidth: 1,
    borderRadius: 10,
    borderColor: '#CCC',
    padding: 10,
    fontSize: 16,
    marginRight: 10,
    textAlignVertical: 'top'
  },
  textInputButton: {
    flexShrink: 1,
  }
});

export default ViewExample;
```

**Comment by ismailcaferoglu** [comment]
Hi @ardaogulcan,

I tried the solution in your last comment about this issue but I still doesn't work. 

My codes are as below. How can I fix it. 

            <View style={{ flex: 1, backgroundColor: '#DADFE1' }}>
                <View style={{ flex: 1 }}>

                    <View style={{ alignItems: 'center', justifyContent: 'center', flexDirection: 'column', marginTop: 20 }}>
                        <Text style={{ color: 'black', fontSize: 14 }}>{this.state.object[0].customerName}-{this.state.object[0].issueType}</Text>
                    </View>

                    <View>
                        <FlatList
                            style={{ marginTop: 20, backgroundColor: '#DADFE1' }}
                            data={this.state.object[0].messages}
                            renderItem={this.renderItem}
                            keyExtractor={(item) => item.id}
                        />
                    </View>

                </View>

                <KeyboardAccessoryView alwaysVisible={true}>
                    <View style={{
                        padding: 8,
                        flexDirection: 'row',
                        justifyContent: 'space-between',
                        alignItems: 'center',
                    }}>
                        <TextInput
                            underlineColorAndroid="transparent"
                            style={{
                                flexGrow: 1,
                                borderWidth: 1,
                                borderRadius: 10,
                                borderColor: '#ccc',
                                padding: 10,
                                fontSize: 16,
                                marginRight: 8,
                                textAlignVertical: 'top'
                            }}
                            multiline={true} />{this.renderInputAccessory}
                        <TouchableOpacity onPress={() => alert('Gödnerildi.')}
                            style={{
                                backgroundColor: '#26C281',
                                justifyContent: 'center',
                                borderColor: '#26C281',
                                borderWidth: 1,
                                borderRadius: 25,
                                height: 35,
                                width: 35,

                                alignItems: 'center'
                            }}>
                            <Icon
                                name='send'
                                size={20}
                                color={'#fff'}
                            />
                        </TouchableOpacity>
                    </View>
                </KeyboardAccessoryView>
            </View>

**Comment by ardaogulcan** [comment]
@ismailcaferoglu, I have refactored your code little bit because of your dynamic data. But main structure is same. It is working normally on GenyMotion, `8.0 - API 26`. Probably your problem is somewhere else for maybe in `{this.renderInputAccessory}`

```js
import React, { Component } from 'react';

import {
  StyleSheet,
  View,
  TextInput,
  Button,
  ScrollView,
  Text,
  TouchableOpacity,
  FlatList,
} from 'react-native';

import { KeyboardAccessoryView } from 'eact-native-keyboard-accessory';

const data = Array(50).fill('').map((item, index) => `dummy ${index}`);

class ViewExample extends Component {
  renderItem = ({ item }) => <Text>{item}</Text>

  render() {
    return (
      <View style={{ flex: 1, backgroundColor: '#DADFE1' }}>
        <View style={{ flex: 1 }}>
            <View style={{ alignItems: 'center', justifyContent: 'center', flexDirection: 'column', marginTop: 20 }}>
                <Text style={{ color: 'black', fontSize: 14 }}>test</Text>
            </View>
            <View>
              <FlatList
                style={{ backgroundColor: '#DADFE1' }}
                data={data}
                renderItem={this.renderItem}
                keyExtractor={(item, index) => index}
              />
            </View>
        </View>

        <KeyboardAccessoryView alwaysVisible={true}>
            <View style={{
                padding: 8,
                flexDirection: 'row',
                justifyContent: 'space-between',
                alignItems: 'center',
            }}>
                <TextInput
                    underlineColorAndroid="transparent"
                    style={{
                        flexGrow: 1,
                        borderWidth: 1,
                        borderRadius: 10,
                        borderColor: '#ccc',
                        padding: 10,
                        fontSize: 16,
                        marginRight: 8,
                        textAlignVertical: 'top'
                    }}
                    multiline={true} />
                <TouchableOpacity onPress={() => alert('Gödnerildi.')}
                    style={{
                        backgroundColor: '#26C281',
                        justifyContent: 'center',
                        borderColor: '#26C281',
                        borderWidth: 1,
                        borderRadius: 25,
                        height: 35,
                        width: 35,

                        alignItems: 'center'
                    }}>
                </TouchableOpacity>
            </View>
        </KeyboardAccessoryView>
      </View>
    );
  }
}

export default ViewExample;
```

**Comment by ardaogulcan** [comment]
Seems like issue occurs only on ejected apps. Trying to find a workaround.

**Comment by ismailcaferoglu** [comment]
Hi @ardaogulcan,

I have to finish my project and find a workaround untill next release. You mentioned using scroll view in your last comments in order to fix this problem. I applied it but saw same problem again. Can you write some example code about scroll view usage with keyboard accessory. Maybe I can solve it temporarly.

**Comment by ardaogulcan** [comment]
@ismailcaferoglu for a quick fix u can update `KeyboardAccessoryView.js` in your local `node_modules/react-native-keyboard-accessory`. Change accessory style with the following code,

```js
accessory: {
  position: 'absolute',
  top: 0,
  right: 0,
  left: 0,
  backgroundColor: '#EFF0F1',
},
```

**Comment by kimho** [comment]
@ardaogulcan Thank you! You saved my day!!

**Comment by stantoncbradley** [comment]
you don't need to hack the lib to update the style, just pass the style attribute:
```
  <KeyboardAccessoryNavigation
          nextHidden
          previousHidden
          onDone={() => Keyboard.dismiss()}
          style={{ top: -200 }
        />
```
also for our use case, 0 didn't work, but -200 seems to work across the android devices we tested

**Comment by ardaogulcan** [comment]
Hey everybody, sorry for the late fix. Now, this issue should be fixed with [v0.1.3](https://github.com/ardaogulcan/react-native-keyboard-accessory/releases/tag/0.1.3)
